File: CarShop/CarShop.Common/Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CarShop.Common.Models
{
    public class Order
    {
        [Key]
        [Required]
        public int Id { get; set; }

        [Required]
        public double TotalAmount { get; set; }

        [Required]
        public DateTime Date { get; set; }

        [Required]
        public OrderStatuses OrderStatus { get; set; }

        [Required]
        public int UserId { get; set; }

        [ForeignKey("UserId")]
        public virtual User User { get; set; }

        public virtual List<Car> Cars { get; set; }

        public enum OrderStatuses
        {
            InProgress,
            Paid
        }
    }
}

File: CarShop/CarShop/CarShopClient/src/app/models/Order.ts
import { Car } from "./Car";
import { OrderStatuses } from "./enums/OrderStatuses";
import { User } from "./user";


export interface Order {
    id: number

    totalAmount: number

    date: Date

    orderStatus: OrderStatuses

    userId: number

    user: User

    cars: Car[]
}

